### PR TITLE
test(unit): add unit tests for KVM and SOL

### PR DESCRIPTION
**Pull request by chethanbellechalu-intel**

### CODE REVIEW

File: projects/sol/src/lib/sol.component.spec.ts [hunk 1]
```diff
@@ -22,13 +22,29 @@ describe('SolComponent', () => {
...
+  }))
 
   it('should create', () => {
     expect(component).toBeTruthy()
   })
+
+  it('should clear the core objects on sol stop', () => {
+    component.stopSol()
+
+    expect(component.term).toBe(null)
```
Comment (rsdmike) [inline projects/sol/src/lib/sol.component.spec.ts:42]: This is missing a lot of expectations:
what if redirector is null
how do you know the redirector.stop method was called?
how do know that the terminal was reset method was called and that it was disposed?
what about redirector, dataProcessor and term being set to null?

File: projects/sol/src/lib/sol.component.spec.ts [hunk 2]
```diff
@@ -22,13 +22,29 @@ describe('SolComponent', () => {
...
+    component.mpsServer = 'wss://localhost'
+    component.authToken = 'authToken'
+    tick(4500)
     fixture.detectChanges()
-  })
+    flush()
+  }))
 
   it('should create', () => {
     expect(component).toBeTruthy()
```
Comment (rsdmike) [inline projects/sol/src/lib/sol.component.spec.ts:36]: should create should focus on ensuring all variables are instantiated and defined to what we expect.

all of the following should have an expectation for what they are so the test break if they change:
```
terminal: any
  container!: any
  term: any
  redirector: any
  dataProcessor: any
  deviceState = 0
  logger: ConsoleLogger = new ConsoleLogger(LogLevel.ERROR)
  @Output() deviceStatus: EventEmitter<number> = new EventEmitter<number>()
  @Input() deviceConnection: EventEmitter<boolean> = new EventEmitter<boolean>()
  @Input() public mpsServer = ''
  @Input() public authToken = ''
  @Input() public deviceId = ''
```

File: projects/sol/src/lib/sol.component.spec.ts [hunk 3]
```diff
@@ -22,13 +22,29 @@ describe('SolComponent', () => {
...
+  it('should clear the core objects on sol stop', () => {
+    component.stopSol()
+
+    expect(component.term).toBe(null)
+  })
+
+  it('should update the terminal state on terminal state change', () => {
+    const state = 0
+    component.onTerminalStateChange(component.redirector, state)
+    expect(component.deviceState).toEqual(state)
```
Comment (rsdmike) [inline projects/sol/src/lib/sol.component.spec.ts:48]: Missing expectation that the device status was emitted.

**Review by rsdmike — CHANGES_REQUESTED** [review]
Overall, we need to be sure we are focusing on ensuring the correct result and behavior rather than just ensuring the code coverage number is up. If we focus on just the code coverage number, it doesn't really mean much. I've commented on SOL, but apply the same concepts to KVM as well.

File: projects/sol/src/lib/sol.component.spec.ts (hunk 1 above)
Comment (chethanbellechalu-intel) [reply projects/sol/src/lib/sol.component.spec.ts]: Updated the expectations.

File: projects/sol/src/lib/sol.component.spec.ts (hunk 2 above)
Comment (chethanbellechalu-intel) [reply projects/sol/src/lib/sol.component.spec.ts:38]: Updated

File: projects/sol/src/lib/sol.component.spec.ts (hunk 3 above)
Comment (chethanbellechalu-intel) [reply projects/sol/src/lib/sol.component.spec.ts:66]: Updated

File: projects/kvm/src/lib/kvm.component.spec.ts [hunk 4]
```diff
@@ -15,13 +17,81 @@ describe('KvmComponent', () => {
       .compileComponents()
   })
 
-  beforeEach(() => {
+  beforeEach(fakeAsync(() => {
     fixture = TestBed.createComponent(KvmComponent)
     component = fixture.componentInstance
+    component.mpsServer = 'wss://localhost'
```
Comment (rsdmike) [inline projects/kvm/src/lib/kvm.component.spec.ts:23]: im not sure we should do this in the beforeEach -- the default state is that mpsServer and authToken are empty strings. So i think we should leave that alone, and change them later in the tests where we need it.

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: projects/kvm/src/lib/kvm.component.spec.ts (hunk 4 above)
Comment (chethanbellechalu-intel) [reply projects/kvm/src/lib/kvm.component.spec.ts:29]: updated

**Review by rsdmike — APPROVED** [review]

**Pull request merged**